builtInFunctions/multiESDTNFTTransfer_test.go

```diff
@@ -1193,3 +1193,221 @@ func runMultiTransfer(t *testing.T, isScToScEventLogEnabled bool) (*vmcommon.VMO
+
+	_ = multiTransfer.accounts.SaveAccount(sender)
+	_ = multiTransfer.accounts.SaveAccount(destination)
+	_, _ = multiTransfer.accounts.Commit()
+
+	scCallFunctionAsHex := hex.EncodeToString([]byte("functionToCall"))
+	scCallArg := hex.EncodeToString([]byte("arg"))
+	nonceBytes := big.NewInt(int64(tokenNonce)).Bytes()
+	quantityBytes := big.NewInt(1).Bytes()
+	scCallArgs := [][]byte{[]byte(scCallFunctionAsHex), []byte(scCallArg)}
+	vmInput := &vmcommon.ContractCallInput{
+		VMInput: vmcommon.VMInput{
+			CallValue:   big.NewInt(0),
+			CallerAddr:  senderAddress,
+			Arguments:   [][]byte{destinationAddress, big.NewInt(2).Bytes(), token1, nonceBytes, quantityBytes, token2, big.NewInt(0).Bytes(), quantityBytes},
+			GasProvided: 100000,
+		},
+		RecipientAddr: senderAddress,
+	}
+	vmInput.Arguments = append(vmInput.Arguments, scCallArgs...)
+
+	return vmInput, multiTransfer
+}
+
+func TestESDTNFTMultiTransfer_ProcessBuiltinFunctionOnSameShardWithScCallWithEGLDNotEnabled(t *testing.T) {
+	t.Parallel()
+
+	vmInput, multiTransfer := createSetupForMultiTransferWithEGLD(t)
+	multiTransfer.enableEpochsHandler = &mock.EnableEpochsHandlerStub{
+		IsFlagEnabledCalled: func(flag core.EnableEpochFlag) bool {
+			return flag != EGLDInESDTMultiTransferFlag
+		},
+	}
+
+	sender, err := multiTransfer.accounts.LoadAccount(vmInput.CallerAddr)
+	require.Nil(t, err)
+	destination, err := multiTransfer.accounts.LoadAccount(vmInput.Arguments[0])
+	require.Nil(t, err)
+
+	_, err = multiTransfer.ProcessBuiltinFunction(sender.(vmcommon.UserAccountHandler), destination.(vmcommon.UserAccountHandler), vmInput)
+	require.NotNil(t, err)
+}
+
+func TestESDTNFTMultiTransfer_ProcessBuiltinFunctionOnSameShardWithScCallWithEGLD(t *testing.T) {
+	t.Parallel()
+
+	vmInput, multiTransfer := createSetupForMultiTransferWithEGLD(t)
+	multiTransfer.enableEpochsHandler = &mock.EnableEpochsHandlerStub{
+		IsFlagEnabledCalled: func(flag core.EnableEpochFlag) bool {
+			return true
+		},
+	}
+
+	sender, err := multiTransfer.accounts.LoadAccount(vmInput.CallerAddr)
+	require.Nil(t, err)
+	destination, err := multiTransfer.accounts.LoadAccount(vmInput.RecipientAddr)
+	require.Nil(t, err)
+
+	vmOutput, err := multiTransfer.ProcessBuiltinFunction(sender.(vmcommon.UserAccountHandler), destination.(vmcommon.UserAccountHandler), vmInput)
+	require.Nil(t, err)
+	require.Equal(t, vmcommon.Ok, vmOutput.ReturnCode)
+
+	_ = multiTransfer.accounts.SaveAccount(sender)
+	_, _ = multiTransfer.accounts.Commit()
+
+	// reload accounts
+	sender, err = multiTransfer.accounts.LoadAccount(vmInput.CallerAddr)
+	require.Nil(t, err)
+	destination, err = multiTransfer.accounts.LoadAccount(vmInput.Arguments[0])
+	require.Nil(t, err)
+
+	testNFTTokenShouldExist(t, multiTransfer.marshaller, sender, []byte("token1"), 1, big.NewInt(2)) // 3 initial - 1 transferred
+	testNFTTokenShouldExist(t, multiTransfer.marshaller, destination, []byte("token1"), 1, big.NewInt(4))
```

Comment:
These values are honestly pretty tricky to track unless you know exactly what `createSetupForMultiTransferWithEGLD` does and how it initializes balances. 

Perhaps we can add initial balances to accounts as config struct in `createSetupForMultiTransferWithEGLD` or add some simple comment for that func on how it initializes balances

Reply:
only a comment is enough here.